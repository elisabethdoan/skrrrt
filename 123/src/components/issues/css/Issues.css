#issues {
  text-align: center;
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 3vh;
}

#radioButtons {
  margin-top: 3vh;
  margin-bottom: 3vh;
}

.defaultRadioButton {
  opacity: 0;
}

.newRadioButton {
  position: relative;
  background-color: #eee;
  height: 25px;
  width: 25px;
  display:inline-block;
  top:5px;
  right: 15px;
  border-radius: 50%;
  cursor: pointer;
}

.defaultRadioButton:hover ~ .newRadioButton {
  background-color: #ccc;
}

.defaultRadioButton:checked ~ .newRadioButton {
  background-color: #fc6d26;
}

.newRadioButton:after { 
  content: "";
  position: absolute;
  display: none;
  top: 9px;
  left: 9px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  background: #554488;
}

.defaultRadioButton:checked ~ .newRadioButton:after {
  display: block;
}

.container {
  cursor: pointer;
  margin-right: 15px;
}

#sorting {
  margin-top: 10px;
  margin-bottom: 10px;
  margin-left: 10px;
}


@media only screen and (max-width: 600px) {
  #issues {
    margin-bottom: 30px;
  }
  
  #radioButtons {
    display: flex;
    flex-direction: column;
  }

  .container {
    margin-top: 10px;
  }

  #openSpan {
    margin-left: 10px;
  }

  #allSpan {
    margin-left: 27px;
  }
}