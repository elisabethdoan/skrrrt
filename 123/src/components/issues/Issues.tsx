import React from 'react';
import Issue from './Issue';
import './css/Issues.css';
import IssueContext from './IssueContextProvider';

type Checked = 'allIssues' | 'openIssues' | 'closedIssues' | null;

interface Istate {
  checked: Checked;
}

class Issues extends React.Component<{}, Istate> {
  constructor(props: {}) {
    super(props);
    this.state = {
      checked: null,
    };
  }

  /**
   * Retrieves active radiobutton from localstorage and updates checked state
   * based on what was retrieved in localstorage. If nothing was set in localstorage allIssues
   * will be checked by default.
   */
  componentDidMount() {
    const checked = localStorage.getItem('checkedRadioIssues') as Checked;
    if (checked) {
      // eslint-disable-next-line
      this.setState({ checked: checked });
    } else {
      this.setState({ checked: 'allIssues' });
    }
  }

  /**
   * Updates state for checked radiobutton based on event. Saves selected radiobutton to
   * localstorage.
   *
   * @param e - event that triggered the function.
   */
  handleChangeRadio = (e: React.ChangeEvent<HTMLInputElement>) => {
    const checked = e.target.value as Checked;
    // eslint-disable-next-line
    this.setState({ checked: checked });
    localStorage.setItem('checkedRadioIssues', checked!);
  };

  render() {
    const { checked } = this.state;
    let predicate = '';
    if (checked === 'closedIssues') {
      predicate = 'closed';
    } else {
      predicate = 'open';
    }
    return (
      <div id="issues">
        <h2>Issues: </h2>
        <form id="radioButtons">
          <label className="container" htmlFor="allIssues">
            All Issues
            <input type="radio" className="defaultRadioButton" onChange={(this.handleChangeRadio)} id="allIssues" value="allIssues" name="filterButtons" checked={checked === 'allIssues'} />
            <span id="allSpan" className="newRadioButton" />
          </label>
          <label className="container" htmlFor="openIssues">
            Open Issues
            <input type="radio" className="defaultRadioButton" onChange={this.handleChangeRadio} id="openIssues" value="openIssues" name="filterButtons" checked={checked === 'openIssues'} />
            <span id="openSpan" className="newRadioButton" />
          </label>
          <label className="container" htmlFor="closedIssues">
            Closed Issues
            <input type="radio" className="defaultRadioButton" onChange={this.handleChangeRadio} id="closedIssues" value="closedIssues" name="filterButtons" checked={checked === 'closedIssues'} />
            <span className="newRadioButton" />
          </label>
        </form>

        {/* Maps each issue from state to an individual Issue component.
            If 'allIssues' is not checked it will filter on predicate first. */}
        <IssueContext.Consumer>
          {checked === 'allIssues'
            ? (value) => value.map((issue) => <Issue key={issue.id} issue={issue} />)
            : (value) => value.filter((issue) => issue.state === predicate)
              .map((issue) => <Issue key={issue.id} issue={issue} />)}
        </IssueContext.Consumer>
      </div>
    );
  }
}

Issues.contextType = IssueContext;

export default Issues;
