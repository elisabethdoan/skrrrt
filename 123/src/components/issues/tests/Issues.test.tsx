import React from 'react';
import renderer from 'react-test-renderer';
import Issues from '../Issues';

test('Tests that rendering Issues matches snapshot,', () => {
  const tree = renderer.create(<Issues />).toJSON();
  expect(tree).toMatchSnapshot();
});
