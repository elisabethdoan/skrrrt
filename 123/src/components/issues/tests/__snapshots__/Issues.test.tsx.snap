// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests that rendering Issues matches snapshot, 1`] = `
<div
  id="issues"
>
  <h2>
    Issues: 
  </h2>
  <form
    id="radioButtons"
  >
    <label
      className="container"
      htmlFor="allIssues"
    >
      All Issues
      <input
        checked={true}
        className="defaultRadioButton"
        id="allIssues"
        name="filterButtons"
        onChange={[Function]}
        type="radio"
        value="allIssues"
      />
      <span
        className="newRadioButton"
        id="allSpan"
      />
    </label>
    <label
      className="container"
      htmlFor="openIssues"
    >
      Open Issues
      <input
        checked={false}
        className="defaultRadioButton"
        id="openIssues"
        name="filterButtons"
        onChange={[Function]}
        type="radio"
        value="openIssues"
      />
      <span
        className="newRadioButton"
        id="openSpan"
      />
    </label>
    <label
      className="container"
      htmlFor="closedIssues"
    >
      Closed Issues
      <input
        checked={false}
        className="defaultRadioButton"
        id="closedIssues"
        name="filterButtons"
        onChange={[Function]}
        type="radio"
        value="closedIssues"
      />
      <span
        className="newRadioButton"
      />
    </label>
  </form>
</div>
`;
